### PR TITLE
DL-4105 - Implement the service timeout

### PR DESCRIPTION
**Pull request by jgrath**
# JIRA Story DL-4105

## Checklist

*Reviewee* (Replace with your name)
 - [X]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [X]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [X]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message

*Reviewer* (Replace with your name)
 - [ ]  I've confirmed that every effort has been made to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've confirmed appropriate tests has been included with any code added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [ ]  I've confirmed code was added using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -104,9 +104,9 @@ contact-frontend {
 }
 
 timeoutDialog {
-    enabled = true
-    timeout = 20
-    timeoutCountdown = 10
+  enabled = true
+  timeout = 10
```
Comment (grahampaulcook) [inline conf/application.conf:108]: I believe this should be 15 (for 15 minutes)

File: conf/application.conf [hunk 2]
```diff
@@ -104,9 +104,9 @@ contact-frontend {
...
+  timeoutCountdown = 3
```
Comment (grahampaulcook) [inline conf/application.conf:109]: Need to check but I think the prompt should trigger at 2 minutes before sign-out

**Review by grahampaulcook — CHANGES_REQUESTED** [review]
Could we change these to default to a safe value, just in case we don't have config values for them.

**FrontendAppConfig.scala**
  lazy val timeoutDialogTimeout: Int = configuration.get[Int]("timeoutDialog.timeout")
  lazy val timeoutDialogTimeoutCountdown: Int = configuration.get[Int]("timeoutDialog.timeoutCountdown")

**Comment by grahampaulcook** [comment]
@jgrath Could you rebase this PR please as some things have been merged to master

**Review by alijy — CHANGES_REQUESTED** [review]
- The content on the timeout page needs to be adjusted to adhere to HMRC design
- Add some unit tests for the keep-alive method
- Potential improvement: add a separate controller for keep-alive

File: test/views/SignedOutViewSpec.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+      val application = applicationBuilder().build()
+
+      val view = application.injector.instanceOf[WeSignedYouOutPageView]
+
+      val applyView = view.apply()(fakeRequest, messages)
+
+      val doc = asDocument(applyView)
+
+      doc.title() mustBe "For your security, this service has been reset"
```
Comment (alijy) [inline test/views/SignedOutViewSpec.scala:35]: This test should be: 
`doc.title() mustBe "For your security, we signed you out"`

**Review by alijy — CHANGES_REQUESTED** [review]

**Review by alijy — APPROVED** [review]

**Pull request merged**